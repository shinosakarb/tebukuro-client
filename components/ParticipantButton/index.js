// @flow
import React from 'react'

type Props = {
  hasEventWaitlist: boolean,
}

const ParticipantButton = (props: Props) => (
  <input
    type="submit"
    value={props.hasEventWaitlist ? 'キャンセル待ちに登録' : '参加登録'}
  />
)

export default ParticipantButton
