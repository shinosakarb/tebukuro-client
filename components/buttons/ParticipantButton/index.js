// @flow
import React from 'react'
import Button from '@material/react-button'
import MaterialIcon from '@material/react-material-icon'

type Props = {
  eventId: ?number,
  hasEventWaitlist: boolean,
  onClick: Function,
}

const RegisterButton = (props: { onClick: Function }) => (
  <Button
    icon={<MaterialIcon icon="directions_walk" />}
    onClick={props.onClick}
    outlined
  >
    参加登録
  </Button>
)

const WaitlistedButton = (props: { onClick: Function }) => (
  <Button
    icon={<MaterialIcon icon="playlist_add" />}
    onClick={props.onClick}
    outlined
  >
    キャンセル待ちに登録
  </Button>
)

const ParticipantButton = (props: Props) => {
  const onClickHandler = (e: SyntheticEvent<>) => {
    e.preventDefault()
    props.onClick({ id: props.eventId })
  }

  return (
    <div>
      {props.hasEventWaitlist ?
        <WaitlistedButton onClick={onClickHandler} />
          : <RegisterButton onClick={onClickHandler} />
      }
    </div>
  )
}

export default ParticipantButton
