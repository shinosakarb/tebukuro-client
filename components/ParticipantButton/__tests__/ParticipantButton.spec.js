import React from 'react'
import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import ParticipantButton from '../index'

describe('ParticipantButton', () => {
  describe('with event has a waitlist', () => {
    it('renders participant button with waitlited message.', () => {
      const wrapper = shallow(<ParticipantButton hasEventWaitlist />)
      const tree = shallowToJson(wrapper)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('with event has no waitlist', () => {
    it('renders participant button with admit message.', () => {
      const wrapper = shallow(<ParticipantButton hasEventWaitlist={false} />)
      const tree = shallowToJson(wrapper)

      expect(tree).toMatchSnapshot()
    })
  })
})
