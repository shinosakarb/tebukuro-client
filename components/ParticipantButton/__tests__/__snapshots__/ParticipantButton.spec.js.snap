// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantButton with event has a waitlist renders participant button with waitlited message. 1`] = `
<input
  type="submit"
  value="キャンセル待ちに登録"
/>
`;

exports[`ParticipantButton with event has no waitlist renders participant button with admit message. 1`] = `
<input
  type="submit"
  value="参加登録"
/>
`;
